src/arithmetic/arithmetic_expression.c

```diff
@@ -912,6 +879,45 @@ SIValue AR_CASEWHEN(SIValue *argv, int argc) {
 	return d;
 }
 
+SIValue _AR_NodeDegree(SIValue *argv, int argc, GRAPH_EDGE_DIR dir) {
+	assert(argc >= 1 && SI_TYPE(argv[0]) == T_NODE);
+
+	Node *n = (Node *)argv[0].ptrval;
+	Edge *edges = array_new(Edge, 0);
+	GraphContext *gc = GraphContext_GetFromTLS();
+
+	if(argc > 1) {
+		// We're interested in specific relationship type(s).
+		for(int i = 1; i < argc; i++) {
+			// relationship type should be specified as a string.
+			assert(SI_TYPE(argv[i]) & SI_STRING);
+			const char *label = argv[i].stringval;
+
+			// Make sure relationship exists.
+			Schema *s = GraphContext_GetSchema(gc, label, SCHEMA_EDGE);
+			if(!s) continue;
+
+			// Accumulate edges.
+			Graph_GetNodeEdges(gc->g, n, dir, s->id, &edges);
+		}
+	} else {
+		// Get all relations, regardless of their type.
+		Graph_GetNodeEdges(gc->g, n, dir, GRAPH_NO_RELATION, &edges);
+	}
+
+	SIValue res = SI_LongVal(array_len(edges));
+	array_free(edges);
+	return res;
+}
+
+SIValue AR_INCOMEDEGREE(SIValue *argv, int argc) {
+	return _AR_NodeDegree(argv, argc, GRAPH_EDGE_DIR_INCOMING);
+}
+
+SIValue AR_OUTGOINGDEGREE(SIValue *argv, int argc) {
+	return _AR_NodeDegree(argv, argc, GRAPH_EDGE_DIR_OUTGOING);
+}
```

Comment:
We can optionally also add a `degree` function for `GRAPH_EDGE_DIR_BOTH`, but it's not strictly necessary.

Reply:
I guess if we were to add a simple degree(n) function, then BOTH can be use.